.pre-commit-config.yaml

```diff
@@ -19,3 +19,12 @@ repos:
 
       - id: ruff-format
         types_or: [ python, pyi ]
+
+  - repo: local
+    hooks:
+      - id: pixi-update
+        name: pixi update
+        entry: bash -c 'cd $(git rev-parse --show-toplevel) && pixi update && git add pixi.lock'
```

Comment:
when you run `pixi update` it solves for all dependencies across all environments, and then locks all versions together in `pixi.lock` which is the thing that make reproducibility possible in environments that mix together 'conda' and `pypi` dependencies. all to say, if `pixi update` doesn't finish, it should not be committed, and that's why we bake that into a `pre-commit` hook.